src/main/java/s/volodymyr/onlinebookstore/service/impl/CategoryServiceImpl.java

```
@@ -0,0 +1,54 @@
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import s.volodymyr.onlinebookstore.dto.category.CategoryDto;
+import s.volodymyr.onlinebookstore.dto.category.CreateCategoryRequestDto;
+import s.volodymyr.onlinebookstore.exception.EntityNotFoundException;
+import s.volodymyr.onlinebookstore.mapper.CategoryMapper;
+import s.volodymyr.onlinebookstore.model.Category;
+import s.volodymyr.onlinebookstore.repository.category.CategoryRepository;
+import s.volodymyr.onlinebookstore.service.CategoryService;
+
+@RequiredArgsConstructor
+@Service
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+
+    @Override
+    public List<CategoryDto> findAll(Pageable pageable) {
+        return categoryRepository.findAll(pageable).stream()
+                .map(categoryMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public CategoryDto getCategoryById(Long id) {
+        return categoryMapper.toDto(categoryRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Cannot find category by id " + id)
+        ));
+    }
+
+    @Override
+    public CategoryDto save(CreateCategoryRequestDto categoryRequest) {
+        return categoryMapper
+                .toDto(categoryRepository.save(categoryMapper.toCategory(categoryRequest)));
+    }
+
+    @Override
+    public CategoryDto update(Long id, CreateCategoryRequestDto requestDto) {
+        Category category = categoryRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can`t find category by id " + id)
+        );
+        categoryMapper.updateCategory(requestDto, category);
+        return categoryMapper.toDto(categoryRepository.save(category));
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        categoryRepository.deleteById(id);
```

what if there is no category with such id?

---

Here's direct quote from this method's documentation: 
"Deletes the entity with the given id.
If the entity is not found in the persistence store it is silently ignored.
Params:
id – must not be null.
Throws:
IllegalArgumentException – in case the given id is null"

---

@VolodymyrSkvira yup, you're right